draft-ietf-tls-tls13.md

```diff
@@ -4995,22 +5000,24 @@ by IANA:
 This section provides a summary of the legal state transitions for the
 client and server handshakes.  State names (in all capitals, e.g.,
 START) have no formal meaning but are provided for ease of
-comprehension.  Messages which are sent only sometimes are indicated
-in \[].
-
+comprehension.  Actions which are taken only in certain circumstances are
+indicated in []. The notation "K_{send,recv} = foo" means "set the send/recv
+key to the given key".
 
 ## Client
 
 ~~~~
                            START <----+
             Send ClientHello |        | Recv HelloRetryRequest
+       [K_send = early data] |        |
          /                   v        |
-        |                  WAIT_SH ---+
-    Can |                    | Recv ServerHello
-   send |                    V
-  early |                 WAIT_EE
-   data |                    | Recv EncryptedExtensions
-        |           +--------+--------+
+        |                 WAIT_SH ----+
+        |                    | Recv ServerHello
+        |                    | K_send = K_recv = handshake
```

Comment:
Here my previous point holds.  You set K_send after receiving the Finished, not here.